inference/core/models/object_detection_base.py

```diff
@@ -223,6 +230,13 @@ def preprocess(
                 if FIX_BATCH_SIZE or fix_batch_size
                 else 0
             )
+            if batch_padding < 0:
+                raise ValueError(
+                    f"Requested fix_batch_size but passed in {img_in.shape[0]} images "
+                    f"when the model's batch size is {MAX_BATCH_SIZE}\n"
+                    f"Consider turning off fix_batch_size, changing `MAX_BATCH_SIZE` in"
```

Comment:
When would you want to pass `fix_batch_size`? Does that lead to faster speeds? Is there a performance penalty compared to variable batch size if you are passing in less than the fixed size of images?

(eg is inferring on 1 image with bs 1 faster than 1 image with bs 8?)

Reply:
I think there are some cases where the model will error if you feed it less than the fixed batch size. But @paulguerrie has more context on this and I'm sort of just guessing from expereince.

Reply:
Fix batch size was added for using batching + trt. When using TRT, the model has to be compiled for a specific batch size. So it's best to pass `fix_batch_size=True` so that you don't have to recompile the model everytime your batch size changes. In the case of BNSF, they are sending batches of individual letters to stage 2. The number of letters changes all the time but is never more than X. So they set MAX BATCH SIZE to X and fix_batch_size to true. This makes it so that the model is compiled once for a batch size of X.

For GPU, this isn't necessary as the batch size is truly dynamic. Maybe we should default fix_batch_size to true for our TRT containers?

Reply:
> Maybe we should default fix_batch_size to true for our TRT containers?

And that'd be the BS of the first batch request that came in? Or we'd always set it to MAX?

An alternative could be we rebuild a bigger one for the biggest batch size that gets sent & use the biggest one we have if there's one as big or bigger than the BS already.